src/importer/batch.py

```
@@ -109,9 +128,20 @@ def process_date_range(
 
         # Ensure tables exist
         if ensure_tables:
-            logger.info("Ensuring all tables exist")
+            logger.info("Ensuring all tables exist", data_source=self.data_source.value)
             try:
-                create_all_tables(self.database)
+                if self.data_source == DataSource.NAR:
+                    # Create NAR tables
+                    from src.database.schema_nar import get_nar_schemas
+                    nar_schemas = get_nar_schemas()
+                    for table_name, schema_sql in nar_schemas.items():
```

Ruff (0.14.14)

[warning] Loop control variable `table_name` not used within loop body

Rename unused `table_name` to `_table_name`

(B007)